src/documentation/sdks/python-streaming.md

```diff
@@ -0,0 +1,151 @@
+[DirectRunner]({{ site.baseurl }}/documentation/runners/direct/#streaming-execution)
+and [DataflowRunner]({{ site.baseurl }}/documentation/runners/dataflow/#streaming-execution).
+
+
+## Why use streaming execution?
+
+Beam creates an unbounded PCollection if your pipeline reads from a streaming or
+continously-updating data source (such as Cloud Pub/Sub or Kafka). A runner must
+process an unbounded PCollection using a streaming job that runs continuously,
+as the entire collection is never available for processing at any one time.
+[Size and boundedness]({{ site.baseurl }}/documentation/programming-guide/#size-and-boundedness)
+has more information about bounded and unbounded collections.
+
+
+## Modifying a pipeline to use stream processing
+
+To modify a batch pipeline to support streaming, you must make the following
+code changes:
+
+* Use an I/O connector that supports reading from an unbounded source.
+* Use an I/O connector that supports writing to an unbounded source.
+* Choose a windowing strategy.
+
+The Beam SDK for Python comes with two I/O connectors that support unbounded
+PCollections: Google Cloud Pub/Sub (reading and writing) and Google BigQuery
+(writing).
+
+The following snippets show the changes needed to modify the batch WordCount
+example to support streaming.
+
+These batch WordCount snippets are from
+[wordcount.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/wordcount.py).
+
+```
+  ...
+  lines = p | 'read' >> ReadFromText(known_args.input)
+  ...
+
+  counts = (lines
+            | 'split' >> (beam.ParDo(WordExtractingDoFn())
+                          .with_output_types(six.text_type))
+            | 'pair_with_one' >> beam.Map(lambda x: (x, 1))
+            | 'group' >> beam.GroupByKey()
+            | 'count' >> beam.Map(count_ones))
+  ...
+
+  output = counts | 'format' >> beam.Map(format_result)
+
+  # Write the output using a "Write" transform that has side effects.
+  output | 'write' >> WriteToText(known_args.output)
+```
+
+These streaming WordCount snippets are from
+[streaming_wordcount.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/streaming_wordcount.py).
+This code uses an I/O connector that reads from and writes to an unbounded
+source (Cloud Pub/Sub) and specifies a fixed windowing strategy.
+
+```
+  ...
+  lines = p | beam.io.ReadStringsFromPubSub(topic=known_args.input_topic)
+  ...
+
+  counts = (lines
+            | 'split' >> (beam.ParDo(WordExtractingDoFn())
+                          .with_output_types(six.text_type))
+            | 'pair_with_one' >> beam.Map(lambda x: (x, 1))
+            | beam.WindowInto(window.FixedWindows(15, 0))
+            | 'group' >> beam.GroupByKey()
+            | 'count' >> beam.Map(count_ones))
+
+  ...
+
+  output = counts | 'format' >> beam.Map(format_result)
+
+  # Write to Pub/Sub
+  output | beam.io.WriteStringsToPubSub(known_args.output_topic)
+```
+
+## Running a streaming pipeline
```

Comment:
Should we explain how to create input/output pubsub topics and how to publish messages to those? (These are not related to Beam.)